File: hw/CEDENOR/U2/HW17TestCaseSolution/ComplexOperations/src/test/java/ec/edu/espe/complexoperations/controler/PersonControllerTest.java

package ec.edu.espe.complexoperations.controler;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
 *
 * @author Nahomi Cedeño, codecreators, DCCO-ESPE

 */
public class PersonControllerTest {
    
    public PersonControllerTest() {
    }

    /**
     * Test of computeAgeInDays method, of class PersonController.
     */
    @Test
    public void testComputeAgeInDays() {
        System.out.println("computeAgeInDays");
        LocalDate birthDate = LocalDate.parse("2001-08-07");
        int expResult = 8371; //8361 
        long result = PersonController.computeAgeInDays(birthDate);
        assertEquals(expResult, result);
    }

    /**
     * Test of computeAgeInYears method, of class PersonController.
     */
    @Test
    public void testComputeAgeInYears() {
        System.out.println("computeAgeInYears");
        LocalDate birthDate = LocalDate.parse("2001-08-07");
        int expResult = 22;
        int result = PersonController.computeAgeInYears(birthDate);
        assertEquals(expResult, result);
    }
    
}

File: hw/bonifaz/u2/HW17TestCaseSolution/ComplexOperationsJS/tests/PersonController.test.js
const PersonController = require('../ec.edu.espe.complexoperations.controler/PersonController');

test('computeAgeInDays', () => {
    const birthDate = new Date("2001-11-13");
    const expResult = 8274; 
    const result = PersonController.computeAgeInDays(birthDate);
    expect(result).toEqual(expResult);
});

test('computeAgeInYears', () => {
    const birthDate = new Date("2001-11-13");
    const expResult = 22;
    const result = PersonController.computeAgeInYears(birthDate);
    expect(result).toEqual(expResult);
});
